cuda_core/cuda/core/experimental/_launch_config.py

```diff
@@ -89,7 +99,18 @@ def __post_init__(self):
 def _to_native_launch_config(config: LaunchConfig) -> driver.CUlaunchConfig:
     _lazy_init()
     drv_cfg = driver.CUlaunchConfig()
-    drv_cfg.gridDimX, drv_cfg.gridDimY, drv_cfg.gridDimZ = config.grid
+    
+    # If cluster is specified, convert grid from cluster units to block units
+    if config.cluster:
```

Comment:
nit: there's already another if branch checking the same cluster condition a few lines below. Could you combine the two blocks of code so that we don't keep checking the same condition?